tests/suites/test_suite_platform.function

```diff
@@ -32,53 +32,58 @@ void sleep_ms(int milliseconds)
     usleep(milliseconds * 1000);
 #endif
 }
-#endif
-
-/* END_HEADER */
-
-/* BEGIN_DEPENDENCIES */
 
-/* END_DEPENDENCIES */
 
-/* BEGIN_CASE depends_on:MBEDTLS_HAVE_TIME */
-void time_get_milliseconds()
+int test_time_delay_milliseconds_once(int delay_ms)
 {
     mbedtls_ms_time_t current = mbedtls_ms_time();
-    (void) current;
-    /* This goto is added to avoid warnings from the generated code. */
-    goto exit;
+    mbedtls_ms_time_t end;
+    mbedtls_ms_time_t elapsed_ms;
+
+    sleep_ms(delay_ms);
+
+    end = mbedtls_ms_time();
+    elapsed_ms = end - current;
+
+    return elapsed_ms > 0 && elapsed_ms >= delay_ms && \
+           elapsed_ms < 4000 + delay_ms;
 }
-/* END_CASE */
 
-/* BEGIN_CASE depends_on:MBEDTLS_HAVE_TIME */
-void time_get_seconds()
+int test_time_delay_seconds_once(int delay_secs)
 {
     mbedtls_time_t current = mbedtls_time(NULL);
-    (void) current;
-    /* This goto is added to avoid warnings from the generated code. */
-    goto exit;
+    mbedtls_time_t end;
+    mbedtls_time_t elapsed_secs;
+
+    sleep_ms(delay_secs * 1000);
+
+    end = mbedtls_time(NULL);
+    elapsed_secs = end - current;
+
+    return elapsed_secs > 0 && elapsed_secs - delay_secs >= -1 && \
+           elapsed_secs - delay_secs <   4;
+
 }
-/* END_CASE */
+#define MBEDTLS_TIME_DELAY_TEST_RETRIES 3
+#endif
+/* END_HEADER */
+
+/* BEGIN_DEPENDENCIES */
+
+/* END_DEPENDENCIES */
 
 /* BEGIN_CASE depends_on:MBEDTLS_HAVE_TIME */
 void time_delay_milliseconds(int delay_ms)
 {
-    mbedtls_ms_time_t current = mbedtls_ms_time();
-    mbedtls_ms_time_t elapsed_ms;
-
-    /*
-     * WARNING: DO NOT ENABLE THIS TEST. We keep the code here to document the
-     *          reason.
-     *
-     *          Windows CI reports random test fail on platform-suite. It might
-     *          be caused by this case.
-     */
-    sleep_ms(delay_ms);
-
-    elapsed_ms = mbedtls_ms_time() - current;
-    TEST_ASSERT(elapsed_ms >= delay_ms && elapsed_ms < 4000 + delay_ms);
-    /* This goto is added to avoid warnings from the generated code. */
-    goto exit;
+    int i;
+    unsigned flag = 0;
+    for (i = 0; i < MBEDTLS_TIME_DELAY_TEST_RETRIES; i++) {
+        if (test_time_delay_milliseconds_once(delay_ms)) {
+            break;
+        }
+        flag |= 1 << i;
```

Comment:
I don't understand the complex logic with the flag. Why not `goto exit` on success, failure on exiting the for loop normally?

(Maybe moot since I don't like these iterations at all.)

Reply:
>Why not goto exit on success, failure on exiting the for loop normally?

Address Tom's comment

Reply:
reverted.